### PR TITLE
overlay: use xattrs perms with network FSs

### PR DESCRIPTION
**Pull request by giuseppe**
when running in a user namespace on a network file system, automatically switch on the permission xattrs feature so that users
won't have to tweak their storage.conf file with difficult to guess options.
    
The list of remote file systems was taken from coreutils stat:

https://git.savannah.gnu.org/gitweb/?p=coreutils.git;a=blob;f=src/stat.c;h=0c34501e34410b9096edf377654b9ff27fe9fa24;hb=HEAD#l260

Also fix an issue with "silly renames" on NFS, more details in the commit log.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: store.go [hunk 1]
```diff
@@ -2498,7 +2499,15 @@ func (s *store) DeleteContainer(id string) error {
 			gcpath := filepath.Join(s.GraphRoot(), middleDir, container.ID)
 			wg.Add(1)
 			go func() {
-				errChan <- os.RemoveAll(gcpath)
+				var err error
+				for attempts := 0; attempts < 50; attempts++ {
+					err = os.RemoveAll(gcpath)
+					if err == nil || !errors.Is(err, unix.EBUSY) {
```
Comment (flouthoc) [inline store.go:2505]: I think we need move `erros.Is(err, unix.EBUSY)` with a wrapper to unix specific package with `// +build linux freebsd darwin` preferably https://github.com/containers/storage/blob/main/pkg/system/syscall_unix.go and return no-op for non-unix.

File: drivers/overlay/overlay.go [hunk 2]
```diff
@@ -266,18 +283,27 @@ func Init(home string, options graphdriver.Options) (graphdriver.Driver, error)
 	}
 
 	if opts.mountProgram != "" {
+		if unshare.IsRootless() && isNetworkFileSystem(fsMagic) && opts.forceMask == nil {
+			m := os.FileMode(0700)
+			opts.forceMask = &m
+			logrus.Debugf("Network file system detected as backing store.  Enforcing overlay option `force_mask=\"%o\"`.", m)
```
Comment (rhatdan) [inline drivers/overlay/overlay.go:289]: This should be at the Info level, or should it be Warn?  Would it be better for users to configure this themselves within the storage.conf, and warn them that we did it automatically.

Comment (giuseppe) [reply drivers/overlay/overlay.go:289]: I had it at Warn level first, but seemed annoying.  I can switch it back to Warnf

Comment (giuseppe) [reply drivers/overlay/overlay.go]: reverted to Warning

File: store.go (hunk 1 above)
Comment (giuseppe) [reply store.go]: thanks, fixed now

**Comment by rhatdan** [comment]
We can see if the Warnf is too agravating, or is easy enough to fix.

**Review by rhatdan — APPROVED** [review]
LGTM

**Comment by rhatdan** [comment]
@flouthoc @saschagrunert @nalind @TomSweeneyRedHat @ashley-cui @umohnani8 PTAL

**Review by flouthoc — APPROVED** [review]

**Comment by flouthoc** [comment]
LGTM, Restarted failing test but not sure if it was a flake.

**Pull request merged**